In [1]:
!python -m pip install -U python-woc pandas matplotlib
# please clear the output of this cell in your notebook before checking it in

Defaulting to user installation because normal site-packages is not writeable


In [2]:
from woc.remote import WocMapsRemote
from tqdm import tqdm
import pandas as pd
# creates the client
woc = WocMapsRemote( base_url="https://worldofcode.org/api/")
# if you got an API key
# woc = WocMapsRemote( base_url="https://worldofcode.org/api/", api_key="woc-XXXXXX-YYYYYY" )

# Now for each of the ten projects you were assigned get commits, e.g.
projects = ['nwfsc-cb/OCNMS', 'skuschel/postpic', 'florianhartig/DHARMa', 'Acellera/moleculekit', 'druid-io/pydruid', 'projectchrono/chrono',
            'MolSSI/QCEngine', 'Radiomics/pyradiomics', 'jalammar/ecco', 'pyscf/pyscf',]
list_df_commits = []
for prj in projects:
  # Convert GitHub repository names to WoC V2412 project names
  prj = prj.lower().replace('/','_',2)
  commits = woc.get_values('p2c', prj)
  df = pd.DataFrame(commits, columns=["sha1"])
  df['project'] = prj
  list_df_commits.append(df)
df = pd.concat(list_df_commits)
#perhaps save the list (if no errors) so you do not need to retrieve them again
df.to_csv('df_commits.csv')
df.head(1)

,sha1,project
0,00a981cbd61fe7e318a3afee58ab02787f864990,nwfsc-cb_ocnms


In [3]:
# If the number of commits its not very large, you can try to get them all at the same time,
# The max batch size is 10
import time
# let us first split df['sha1'] in chunks
chunks = [df['sha1'][x:x+10] for x in range(0, len(df), 10)]
commit_data = []

for chunk in tqdm(chunks): # iterate over the commits
  # res, err = woc.show_content_many('commit',chunk.to_list())

  # commit.tch returns the same data but faster
  res, err = woc.get_values_many('commit.tch',chunk.to_list())
  res = {k: v[0] for k, v in res.items()}  # this conversion is necessary because of the internal implementation

  # to walk around the rate limit
  time.sleep(1)

  if err: # check for errors
    print('Got Errors', err)

  for commit_sha, commit in res.items():
    # flatten commit objects
    commit_data.append({
          'commit': commit_sha,
          'tree': commit[0],
          'parent': list(commit[1]),
          'author': commit[2][0],
          'author_time': int(commit[2][1]),
          'author_tz': commit[2][2],
          'committer': commit[3][0],
          'committer_time': int(commit[3][1]),
          'committer_tz': commit[3][2],
          'message': commit[4],
    })

df_commit_data = pd.DataFrame(commit_data)
df_commit_data = df_commit_data.merge(df, left_on='commit', right_on='sha1')
df_commit_data.to_csv('df_commit_data.csv', index=False)
df_commit_data.head(2)

  1%|▉                                                                             | 56/4785 [00:57<1:20:55,  1.03s/it]

Got Errors {'0498ba3fc38386c134ead3a3eb89d984d7ba2ae0': 'Key 0498ba3fc38386c134ead3a3eb89d984d7ba2ae0 not found in /da5_fast/All.sha1c/commit_4.tch'}


  2%|█▎                                                                            | 82/4785 [01:24<1:20:20,  1.02s/it]

Got Errors {'395e21365b03819bea47d53ac23cf49ccb2d58e6': 'Key 395e21365b03819bea47d53ac23cf49ccb2d58e6 not found in /da5_fast/All.sha1c/commit_57.tch'}


  2%|█▊                                                                           | 110/4785 [01:52<1:19:39,  1.02s/it]

Got Errors {'761223ce5f37b3b92971b017005b96548e232702': 'Key 761223ce5f37b3b92971b017005b96548e232702 not found in /da5_fast/All.sha1c/commit_118.tch'}


  2%|█▉                                                                           | 118/4785 [02:01<1:19:29,  1.02s/it]

Got Errors {'85e199e7d6eddff865e07e8c757662fb6b81f82e': 'Key 85e199e7d6eddff865e07e8c757662fb6b81f82e not found in /da5_fast/All.sha1c/commit_5.tch'}


  3%|██▎                                                                          | 147/4785 [02:30<1:18:59,  1.02s/it]

Got Errors {'c18bf1de682e24be2c23d94d83ed1f4ebf8a1c49': 'Key c18bf1de682e24be2c23d94d83ed1f4ebf8a1c49 not found in /da5_fast/All.sha1c/commit_65.tch'}


  3%|██▌                                                                          | 157/4785 [02:40<1:18:50,  1.02s/it]

Got Errors {'d5a674dbe7e31b93e5e7a50d6e53313262d0d35f': 'Key d5a674dbe7e31b93e5e7a50d6e53313262d0d35f not found in /da5_fast/All.sha1c/commit_85.tch'}


  3%|██▌                                                                          | 159/4785 [02:42<1:18:49,  1.02s/it]

Got Errors {'d8f54693b239f25848cc7fc33d0578452e95f09e': 'Key d8f54693b239f25848cc7fc33d0578452e95f09e not found in /da5_fast/All.sha1c/commit_88.tch'}


  4%|██▋                                                                          | 170/4785 [02:54<1:18:34,  1.02s/it]

Got Errors {'f22a53493c6b0b5429bfefca695c143af4b477cd': 'Key f22a53493c6b0b5429bfefca695c143af4b477cd not found in /da5_fast/All.sha1c/commit_114.tch', 'f2622f67320210b00cbc09f04949164c1a033d28': 'Key f2622f67320210b00cbc09f04949164c1a033d28 not found in /da5_fast/All.sha1c/commit_114.tch'}


  6%|█████                                                                        | 311/4785 [05:19<1:16:46,  1.03s/it]

Got Errors {'094c21dc7e06a954ed49b8faef1a03235a95ea8b': 'Key 094c21dc7e06a954ed49b8faef1a03235a95ea8b not found in /da5_fast/All.sha1c/commit_9.tch', '0a16df4824c06d41abb0f016d74e56efa60af823': 'Key 0a16df4824c06d41abb0f016d74e56efa60af823 not found in /da5_fast/All.sha1c/commit_10.tch'}


  7%|█████▏                                                                       | 320/4785 [05:28<1:16:30,  1.03s/it]

Got Errors {'18f9d85250d43c2a75fe713a5e8506a425c83a22': 'Key 18f9d85250d43c2a75fe713a5e8506a425c83a22 not found in /da5_fast/All.sha1c/commit_24.tch'}


  7%|█████▎                                                                       | 330/4785 [05:38<1:15:58,  1.02s/it]

Got Errors {'293dd94f3c672f80c7175b99e8c72b51d504a9d1': 'Key 293dd94f3c672f80c7175b99e8c72b51d504a9d1 not found in /da5_fast/All.sha1c/commit_41.tch'}


  7%|█████▍                                                                       | 335/4785 [05:44<1:15:49,  1.02s/it]

Got Errors {'3254fe78f067f31b690b54524595a83fe0830a15': 'Key 3254fe78f067f31b690b54524595a83fe0830a15 not found in /da5_fast/All.sha1c/commit_50.tch'}


  7%|█████▍                                                                       | 337/4785 [05:46<1:15:48,  1.02s/it]

Got Errors {'3541601fe72694bce597ecb0fb8aa1e811cb4cc8': 'Key 3541601fe72694bce597ecb0fb8aa1e811cb4cc8 not found in /da5_fast/All.sha1c/commit_53.tch'}


  7%|█████▍                                                                       | 340/4785 [05:49<1:15:44,  1.02s/it]

Got Errors {'3ad026629838b0f79d318fbe65b8baa54ccda433': 'Key 3ad026629838b0f79d318fbe65b8baa54ccda433 not found in /da5_fast/All.sha1c/commit_58.tch'}


  7%|█████▌                                                                       | 342/4785 [05:51<1:15:42,  1.02s/it]

Got Errors {'3e3928958067c47a27e650fad11edd0a443c6b68': 'Key 3e3928958067c47a27e650fad11edd0a443c6b68 not found in /da5_fast/All.sha1c/commit_62.tch'}


  7%|█████▌                                                                       | 345/4785 [05:54<1:15:38,  1.02s/it]

Got Errors {'416fde0a26a96627d5908691efadc48cc9a583b3': 'Key 416fde0a26a96627d5908691efadc48cc9a583b3 not found in /da5_fast/All.sha1c/commit_65.tch'}


  7%|█████▌                                                                       | 347/4785 [05:56<1:15:35,  1.02s/it]

Got Errors {'459260bfe49142bbff2bd5c994323b9e7f2766ca': 'Key 459260bfe49142bbff2bd5c994323b9e7f2766ca not found in /da5_fast/All.sha1c/commit_69.tch'}


  7%|█████▋                                                                       | 352/4785 [06:01<1:15:20,  1.02s/it]

Got Errors {'4d27293ade64fb65750b94a17888817f63d18c7e': 'Key 4d27293ade64fb65750b94a17888817f63d18c7e not found in /da5_fast/All.sha1c/commit_77.tch'}


  7%|█████▋                                                                       | 353/4785 [06:02<1:15:23,  1.02s/it]

Got Errors {'50f87a3f89b9c5eb2f0e2fe4e7a76db1f0f6b137': 'Key 50f87a3f89b9c5eb2f0e2fe4e7a76db1f0f6b137 not found in /da5_fast/All.sha1c/commit_80.tch', '515f62cc6dd6e4d3a3cc3a25015b4f023f7be41d': 'Key 515f62cc6dd6e4d3a3cc3a25015b4f023f7be41d not found in /da5_fast/All.sha1c/commit_81.tch'}


  7%|█████▋                                                                       | 356/4785 [06:05<1:15:20,  1.02s/it]

Got Errors {'561ee6cfc6a40af55c206ce3241b20b214eff1ec': 'Key 561ee6cfc6a40af55c206ce3241b20b214eff1ec not found in /da5_fast/All.sha1c/commit_86.tch'}


  8%|█████▊                                                                       | 360/4785 [06:09<1:15:19,  1.02s/it]

Got Errors {'5c43ee177c63fc7de715a9086e4c00741574283b': 'Key 5c43ee177c63fc7de715a9086e4c00741574283b not found in /da5_fast/All.sha1c/commit_92.tch'}


  8%|█████▊                                                                       | 361/4785 [06:10<1:15:19,  1.02s/it]

Got Errors {'5f6be6ef0c7f2e709095ec49cad32ee0ddb6fccd': 'Key 5f6be6ef0c7f2e709095ec49cad32ee0ddb6fccd not found in /da5_fast/All.sha1c/commit_95.tch'}


  8%|█████▉                                                                       | 369/4785 [06:18<1:16:21,  1.04s/it]

Got Errors {'6c7bfa14138feff5638186640811697bc7058d6d': 'Key 6c7bfa14138feff5638186640811697bc7058d6d not found in /da5_fast/All.sha1c/commit_108.tch'}


  8%|█████▉                                                                       | 371/4785 [06:20<1:15:43,  1.03s/it]

Got Errors {'702c0a7a05f379bb9b50cf54d24c8a828054f51e': 'Key 702c0a7a05f379bb9b50cf54d24c8a828054f51e not found in /da5_fast/All.sha1c/commit_112.tch'}


  8%|██████▍                                                                      | 401/4785 [06:51<1:14:34,  1.02s/it]

Got Errors {'a1295ba3975e8100f8bd3c0e7e22515862b6a048': 'Key a1295ba3975e8100f8bd3c0e7e22515862b6a048 not found in /da5_fast/All.sha1c/commit_33.tch'}


  8%|██████▍                                                                      | 403/4785 [06:53<1:14:29,  1.02s/it]

Got Errors {'a4f9d140f34a10e1a6a086e18683fc3793c58e12': 'Key a4f9d140f34a10e1a6a086e18683fc3793c58e12 not found in /da5_fast/All.sha1c/commit_36.tch'}


  9%|██████▌                                                                      | 407/4785 [06:57<1:14:23,  1.02s/it]

Got Errors {'acb9c1f80d857252e4de0410bbcc4792da42bb1c': 'Key acb9c1f80d857252e4de0410bbcc4792da42bb1c not found in /da5_fast/All.sha1c/commit_44.tch'}


  9%|██████▌                                                                      | 408/4785 [06:58<1:14:26,  1.02s/it]

Got Errors {'ada57f93745b3c07d2fd548fc769c3cb748d9667': 'Key ada57f93745b3c07d2fd548fc769c3cb748d9667 not found in /da5_fast/All.sha1c/commit_45.tch'}


  9%|██████▋                                                                      | 412/4785 [07:02<1:14:26,  1.02s/it]

Got Errors {'b6e3db59cc5720c032872ef639e4a895e0a534b3': 'Key b6e3db59cc5720c032872ef639e4a895e0a534b3 not found in /da5_fast/All.sha1c/commit_54.tch'}


  9%|██████▊                                                                      | 421/4785 [07:12<1:14:29,  1.02s/it]

Got Errors {'c5e0b5c6a213a8f5263ec0fc95fe6834bd877609': 'Key c5e0b5c6a213a8f5263ec0fc95fe6834bd877609 not found in /da5_fast/All.sha1c/commit_69.tch'}


  9%|██████▊                                                                      | 424/4785 [07:15<1:14:15,  1.02s/it]

Got Errors {'cb585dc21db347bdd6c3b1750bdc27d294da1d01': 'Key cb585dc21db347bdd6c3b1750bdc27d294da1d01 not found in /da5_fast/All.sha1c/commit_75.tch'}


  9%|██████▊                                                                      | 426/4785 [07:17<1:17:35,  1.07s/it]

Got Errors {'cd731f3ea853e6a0126a32528b7e01c08c90e00e': 'Key cd731f3ea853e6a0126a32528b7e01c08c90e00e not found in /da5_fast/All.sha1c/commit_77.tch'}


  9%|██████▉                                                                      | 434/4785 [07:25<1:14:37,  1.03s/it]

Got Errors {'d75956caf8c6cfb4f2a754ac5f220f34819c997d': 'Key d75956caf8c6cfb4f2a754ac5f220f34819c997d not found in /da5_fast/All.sha1c/commit_87.tch'}


  9%|███████                                                                      | 439/4785 [07:30<1:14:08,  1.02s/it]

Got Errors {'e045d6ed8cab58046e3f6317a8c724509566ec2f': 'Key e045d6ed8cab58046e3f6317a8c724509566ec2f not found in /da5_fast/All.sha1c/commit_96.tch'}


  9%|███████                                                                      | 441/4785 [07:32<1:14:03,  1.02s/it]

Got Errors {'e2d4283b9fd7956a75a3debbd5bb3b9fb21ccf1f': 'Key e2d4283b9fd7956a75a3debbd5bb3b9fb21ccf1f not found in /da5_fast/All.sha1c/commit_98.tch'}


  9%|███████▏                                                                     | 449/4785 [07:40<1:13:47,  1.02s/it]

Got Errors {'f2c16ca8707d550332105127ac0c4ab08dbed648': 'Key f2c16ca8707d550332105127ac0c4ab08dbed648 not found in /da5_fast/All.sha1c/commit_114.tch'}


  9%|███████▏                                                                     | 450/4785 [07:41<1:13:47,  1.02s/it]

Got Errors {'f44589306ed0be188f9e8c04848a24695e5e8f05': 'Key f44589306ed0be188f9e8c04848a24695e5e8f05 not found in /da5_fast/All.sha1c/commit_116.tch'}


 10%|███████▍                                                                     | 466/4785 [07:58<1:13:31,  1.02s/it]

Got Errors {'1c33711f53eacd5bd79edd8fe5f5965b2e916457': 'Key 1c33711f53eacd5bd79edd8fe5f5965b2e916457 not found in /da5_fast/All.sha1c/commit_28.tch'}


 10%|███████▋                                                                     | 476/4785 [08:08<1:13:24,  1.02s/it]

Got Errors {'3a70f72844d0ec1657d7198981849080a3b58701': 'Key 3a70f72844d0ec1657d7198981849080a3b58701 not found in /da5_fast/All.sha1c/commit_58.tch'}


 11%|████████▎                                                                    | 519/4785 [08:52<1:12:54,  1.03s/it]

Got Errors {'b336bf969d1b56681cbfb24aabd3c7cf2a1282d9': 'Key b336bf969d1b56681cbfb24aabd3c7cf2a1282d9 not found in /da5_fast/All.sha1c/commit_51.tch'}


 11%|████████▊                                                                    | 544/4785 [09:18<1:12:14,  1.02s/it]

Got Errors {'fbb5d4571e46ba1fc6fcf357bf51116cfe5558d8': 'Key fbb5d4571e46ba1fc6fcf357bf51116cfe5558d8 not found in /da5_fast/All.sha1c/commit_123.tch'}


 11%|████████▊                                                                    | 546/4785 [09:20<1:13:10,  1.04s/it]

Got Errors {'ffc9a97ce2d2a796fb5c5718dbddb924cd3b6c5d': 'Key ffc9a97ce2d2a796fb5c5718dbddb924cd3b6c5d not found in /da5_fast/All.sha1c/commit_127.tch'}


 12%|█████████                                                                    | 562/4785 [09:36<1:11:55,  1.02s/it]

Got Errors {'01e2578746c7edc76270157d390f9d3c831d3a1b': 'Key 01e2578746c7edc76270157d390f9d3c831d3a1b not found in /da5_fast/All.sha1c/commit_1.tch'}


 17%|█████████████▏                                                               | 820/4785 [14:00<1:07:25,  1.02s/it]

Got Errors {'1f8a7f9a95372493b6f158f204679a98d2dd48e5': 'Key 1f8a7f9a95372493b6f158f204679a98d2dd48e5 not found in /da5_fast/All.sha1c/commit_31.tch'}


 31%|████████████████████████▍                                                     | 1496/4785 [25:34<56:11,  1.02s/it]

Got Errors {'6927a5b99effc19143d9af6ce58d25e03c2a1eaf': 'Key 6927a5b99effc19143d9af6ce58d25e03c2a1eaf not found in /da5_fast/All.sha1c/commit_105.tch'}


 42%|████████████████████████████████▊                                             | 2016/4785 [34:26<47:08,  1.02s/it]

Got Errors {'a4142e77940276e670bc586ae8a04bf369aa87b6': 'Key a4142e77940276e670bc586ae8a04bf369aa87b6 not found in /da5_fast/All.sha1c/commit_36.tch'}


 43%|█████████████████████████████████▊                                            | 2074/4785 [35:26<46:44,  1.03s/it]

Got Errors {'aa5f76507411caa600c075cb20b7a29259685ff9': 'Key aa5f76507411caa600c075cb20b7a29259685ff9 not found in /da5_fast/All.sha1c/commit_42.tch'}


 61%|███████████████████████████████████████████████▎                              | 2900/4785 [49:41<32:43,  1.04s/it]

Got Errors {'2c3b66c6c8f3cec640392ddaa7514891602f6274': 'Key 2c3b66c6c8f3cec640392ddaa7514891602f6274 not found in /da5_fast/All.sha1c/commit_44.tch'}


 62%|████████████████████████████████████████████████▋                             | 2985/4785 [51:10<31:17,  1.04s/it]

Got Errors {'7a77c8dac884abf6e5f769ed14b2136eccc98dc2': 'Key 7a77c8dac884abf6e5f769ed14b2136eccc98dc2 not found in /da5_fast/All.sha1c/commit_122.tch'}


 64%|█████████████████████████████████████████████████▋                            | 3047/4785 [52:15<30:05,  1.04s/it]

Got Errors {'b195e68bbecf7b15b05db0c37baf458d2020a9bc': 'Key b195e68bbecf7b15b05db0c37baf458d2020a9bc not found in /da5_fast/All.sha1c/commit_49.tch'}


 66%|███████████████████████████████████████████████████▋                          | 3168/4785 [54:21<28:08,  1.04s/it]

Got Errors {'32bbe0a9e15ce19f548909d28b54c5244a7c86f8': 'Key 32bbe0a9e15ce19f548909d28b54c5244a7c86f8 not found in /da5_fast/All.sha1c/commit_50.tch'}


 66%|███████████████████████████████████████████████████▋                          | 3171/4785 [54:24<27:59,  1.04s/it]

Got Errors {'36d1136dc1ded8241a27011fd1d8d047777d5a84': 'Key 36d1136dc1ded8241a27011fd1d8d047777d5a84 not found in /da5_fast/All.sha1c/commit_54.tch'}


 66%|███████████████████████████████████████████████████▊                          | 3177/4785 [54:30<27:45,  1.04s/it]

Got Errors {'40b8e08522b6d0a45bf71c0f79776c4edf81d597': 'Key 40b8e08522b6d0a45bf71c0f79776c4edf81d597 not found in /da5_fast/All.sha1c/commit_64.tch'}


 66%|███████████████████████████████████████████████████▊                          | 3179/4785 [54:32<27:43,  1.04s/it]

Got Errors {'43fe880e00594293a8924f28fc42bdb01749732a': 'Key 43fe880e00594293a8924f28fc42bdb01749732a not found in /da5_fast/All.sha1c/commit_67.tch'}


 67%|████████████████████████████████████████████████████                          | 3190/4785 [54:44<27:41,  1.04s/it]

Got Errors {'542c8ce8630f2367d0f924e8bf6db17c47cbf61e': 'Key 542c8ce8630f2367d0f924e8bf6db17c47cbf61e not found in /da5_fast/All.sha1c/commit_84.tch'}


 67%|████████████████████████████████████████████████████▎                         | 3210/4785 [55:05<27:10,  1.04s/it]

Got Errors {'73bb29a4accc6d5346cc5e4f021ad0c89abbc9f7': 'Key 73bb29a4accc6d5346cc5e4f021ad0c89abbc9f7 not found in /da5_fast/All.sha1c/commit_115.tch'}


 67%|████████████████████████████████████████████████████▍                         | 3216/4785 [55:11<27:11,  1.04s/it]

Got Errors {'7cec6a2ebce8d61505616e5942f503cd82b7d3c5': 'Key 7cec6a2ebce8d61505616e5942f503cd82b7d3c5 not found in /da5_fast/All.sha1c/commit_124.tch'}


 67%|████████████████████████████████████████████████████▍                         | 3219/4785 [55:14<27:01,  1.04s/it]

Got Errors {'8138016999aa2afe59f27668c1247510e615bb19': 'Key 8138016999aa2afe59f27668c1247510e615bb19 not found in /da5_fast/All.sha1c/commit_1.tch'}


 68%|█████████████████████████████████████████████████████▏                        | 3261/4785 [55:58<26:18,  1.04s/it]

Got Errors {'bd0504043b55782ecbd02a14eff8e7a700b67613': 'Key bd0504043b55782ecbd02a14eff8e7a700b67613 not found in /da5_fast/All.sha1c/commit_61.tch'}


 68%|█████████████████████████████████████████████████████▎                        | 3270/4785 [56:07<26:11,  1.04s/it]

Got Errors {'ca59d924d3da7433cb96236823abde748c1abe37': 'Key ca59d924d3da7433cb96236823abde748c1abe37 not found in /da5_fast/All.sha1c/commit_74.tch'}


 69%|█████████████████████████████████████████████████████▍                        | 3281/4785 [56:18<26:06,  1.04s/it]

Got Errors {'dd5a76841a1769ab98ba1282147d795a2e72b394': 'Key dd5a76841a1769ab98ba1282147d795a2e72b394 not found in /da5_fast/All.sha1c/commit_93.tch'}


 79%|████████████████████████████████████████████████████████████▎               | 3798/4785 [1:05:16<17:02,  1.04s/it]

Got Errors {'4ff0057dd9e5523aa1fd7fa6e5f7d78488598c9b': 'Key 4ff0057dd9e5523aa1fd7fa6e5f7d78488598c9b not found in /da5_fast/All.sha1c/commit_79.tch'}


 81%|█████████████████████████████████████████████████████████████▎              | 3862/4785 [1:06:22<15:56,  1.04s/it]

Got Errors {'5ad36fc113239ea28428a6122a22f162e25d23b9': 'Key 5ad36fc113239ea28428a6122a22f162e25d23b9 not found in /da5_fast/All.sha1c/commit_90.tch'}


 83%|███████████████████████████████████████████████████████████████▏            | 3976/4785 [1:08:20<13:57,  1.03s/it]

Got Errors {'6fb91346b4ad6f3c019583a69c1c0f98d095072d': 'Key 6fb91346b4ad6f3c019583a69c1c0f98d095072d not found in /da5_fast/All.sha1c/commit_111.tch'}


 87%|█████████████████████████████████████████████████████████████████▉          | 4148/4785 [1:11:18<10:59,  1.03s/it]

Got Errors {'8e6f5decaf3fba8055c7e37a6926819ccf25a8f2': 'Key 8e6f5decaf3fba8055c7e37a6926819ccf25a8f2 not found in /da5_fast/All.sha1c/commit_14.tch'}


 92%|█████████████████████████████████████████████████████████████████████▋      | 4385/4785 [1:15:26<07:00,  1.05s/it]

Got Errors {'b8502e4673bb7739ce3d0aa58ad7215c05958638': 'Key b8502e4673bb7739ce3d0aa58ad7215c05958638 not found in /da5_fast/All.sha1c/commit_56.tch'}


 95%|████████████████████████████████████████████████████████████████████████▌   | 4568/4785 [1:18:38<03:47,  1.05s/it]

Got Errors {'d88630d75a36fec5588e0edda2ecc59b8999c8ef': 'Key d88630d75a36fec5588e0edda2ecc59b8999c8ef not found in /da5_fast/All.sha1c/commit_88.tch'}


 96%|████████████████████████████████████████████████████████████████████████▉   | 4591/4785 [1:19:02<03:22,  1.05s/it]

Got Errors {'dd0817a41b673e7f5587020319f8908be90df074': 'Key dd0817a41b673e7f5587020319f8908be90df074 not found in /da5_fast/All.sha1c/commit_93.tch'}


 97%|█████████████████████████████████████████████████████████████████████████▊  | 4644/4785 [1:19:57<02:26,  1.04s/it]

Got Errors {'e78766d6084b1c3955423b4a39463eaadfe0887c': 'Key e78766d6084b1c3955423b4a39463eaadfe0887c not found in /da5_fast/All.sha1c/commit_103.tch'}


 98%|██████████████████████████████████████████████████████████████████████████▎ | 4680/4785 [1:20:35<01:49,  1.04s/it]

Got Errors {'ee7e5a8f48a8182c6966d54fee36f6043b816a24': 'Key ee7e5a8f48a8182c6966d54fee36f6043b816a24 not found in /da5_fast/All.sha1c/commit_110.tch'}


100%|████████████████████████████████████████████████████████████████████████████| 4785/4785 [1:22:25<00:00,  1.03s/it]


,commit,tree,parent,author,author_time,author_tz,committer,committer_time,committer_tz,message,sha1,project
0,00a981cbd61fe7e318a3afee58ab02787f864990,f2d8ee98870aeeb9e085439e03c5e2f87f781c6a,[6d78feb8e94f93bdf2efb1342f0c98a68b715a8d],Nick Tolimieri <nick.tolimieri@noaa.gov>,1729704027,-0700,Nick Tolimieri <nick.tolimieri@noaa.gov>,1729704027,-0700,Finished initial complete update.\n\nFiles sho...,00a981cbd61fe7e318a3afee58ab02787f864990,nwfsc-cb_ocnms
1,00e5601c2cb1e582978925ef2cdd2521e05eb875,c9b1cb9f53a5614a9c1d5264d97681b62bde952b,[68ffb5d27c84e50b173eb336b13953fda2cf3a4e],jameals <jameal.samhouri@noaa.gov>,1513280110,-0800,jameals <jameal.samhouri@noaa.gov>,1513280110,-0800,updated ms\n,00e5601c2cb1e582978925ef2cdd2521e05eb875,nwfsc-cb_ocnms


In [4]:
# what does the first record looks like?
# commit  parents
v = commit_data[0]
# commit tree, [parent{s}], [author, unixtime, tz ], [ committer, unixtime, tz ], Commit_message ]
print(v)

{'commit': '00a981cbd61fe7e318a3afee58ab02787f864990', 'tree': 'f2d8ee98870aeeb9e085439e03c5e2f87f781c6a', 'parent': ['6d78feb8e94f93bdf2efb1342f0c98a68b715a8d'], 'author': 'Nick Tolimieri <nick.tolimieri@noaa.gov>', 'author_time': 1729704027, 'author_tz': '-0700', 'committer': 'Nick Tolimieri <nick.tolimieri@noaa.gov>', 'committer_time': 1729704027, 'committer_tz': '-0700', 'message': 'Finished initial complete update.\n\nFiles should now run annual update. Have not updated the multivariate file yet.\n'}


In [5]:
# Now that the data is retrieved, save it and commit to your GH fork
# First, we need to flatten info in order to export as csv
# our dataframe will have columns project,'commit, author, time, message'
dfinf = pd.DataFrame({
    'project':  df_commit_data['project'],
    'commit':   df_commit_data['commit'],
    'author':   df_commit_data['author'],
    'time':     df_commit_data['author_time'],
    'message':  df_commit_data['message'],
})


In [6]:
# check if it has the right content
dfinf.head(1)

,project,commit,author,time,message
0,nwfsc-cb_ocnms,00a981cbd61fe7e318a3afee58ab02787f864990,Nick Tolimieri <nick.tolimieri@noaa.gov>,1729704027,Finished initial complete update.\n\nFiles sho...


In [7]:
#mode a means append, so you have all your projects in the same file
yournetid='oyoungbl'
dfinf.to_csv(yournetid+'_project_summary.csv', index=False,sep=';', mode='a', header=False)

# Make sure you check in to your fork not just the notebook but also the csv files!!!

# Don't forget to add requested data from github and this notebook
### For each of the 10 projects go to their github repo and get the number of stars, number of forks, and the last commit date
### Report (in your notebook) the number of commits, the number of authors, and max and min time for each project based on WoC commits and also add the info you obtained from github

In [8]:
df_commit_data['dt'] = pd.to_datetime(df_commit_data['author_time'], unit='s')
woc_stats = df_commit_data.groupby('project').agg(
    commits=('commit', 'nunique'),
    authors=('author', 'nunique'),
    max_time=('dt', 'max'),
    min_time=('dt', 'min'))
woc_stats

,commits,authors,max_time,min_time
project,,,,
acellera_moleculekit,1477,19,2025-10-20 11:34:40,2019-03-28 10:55:31
druid-io_pydruid,891,200,2025-10-09 19:32:30,2013-08-28 00:52:10
florianhartig_dharma,1281,45,2025-10-22 12:01:59,2015-02-10 15:43:53
jalammar_ecco,349,31,2025-09-29 19:20:49,2020-11-07 10:03:22
molssi_qcengine,2868,98,2025-11-05 16:40:37,2018-03-01 22:25:36
nwfsc-cb_ocnms,532,13,2025-08-14 20:32:09,2015-11-16 20:53:30
projectchrono_chrono,23025,266,2025-11-05 14:05:06,2010-02-23 11:03:14
pyscf_pyscf,14445,356,2025-11-05 21:06:40,2014-05-02 18:42:26
radiomics_pyradiomics,1679,58,2025-10-27 22:04:30,2015-09-15 15:19:31


In [9]:
import requests
gh_projects = projects
headers = {}  # optionally {"Authorization": "token YOUR_TOKEN"}
rows = []
for p in gh_projects:
    r = requests.get(f"https://api.github.com/repos/{p}", headers=headers).json()
    c = requests.get(f"https://api.github.com/repos/{p}/commits?per_page=1", headers=headers).json()
    rows.append({'project': p,
                 'stars': r.get('stargazers_count'),
                 'forks': r.get('forks_count'),
                 'last_commit': c[0]['commit']['committer']['date'] if isinstance(c, list) and c else None})
gh_stats = pd.DataFrame(rows)
gh_stats

,project,stars,forks,last_commit
0,nwfsc-cb/OCNMS,1,0,2026-09-25T18:45:26Z
1,skuschel/postpic,69,27,2026-06-08T13:25:46Z
2,florianhartig/DHARMa,265,31,2026-09-25T15:06:13Z
3,Acellera/moleculekit,246,34,2026-09-28T13:33:29Z
4,druid-io/pydruid,520,204,2026-05-04T23:43:54Z
5,projectchrono/chrono,3057,637,2026-09-26T17:45:48Z
6,MolSSI/QCEngine,211,92,2026-08-18T20:32:51Z
7,Radiomics/pyradiomics,1459,549,2025-06-17T00:39:11Z
8,jalammar/ecco,2101,175,2024-08-15T19:08:06Z
9,pyscf/pyscf,1681,749,2026-09-28T05:45:43Z


## Results

### GitHub (stars, forks, last commit date)
| project | stars | forks | last commit |
|---|---|---|---|
| nwfsc-cb/OCNMS | 1 | 0 | 2026-09-25T18:45:26Z |
| skuschel/postpic | 69 | 27 | 2026-06-08T13:25:46Z |
| florianhartig/DHARMa | 265 | 31 | 2026-09-25T15:06:13Z |
| Acellera/moleculekit | 246 | 34 | 2026-09-28T13:33:29Z |
| druid-io/pydruid | 520 | 204 | 2026-05-04T23:43:54Z |
| projectchrono/chrono | 3057 | 637 | 2026-09-26T17:45:48Z |
| MolSSI/QCEngine | 211 | 92 | 2026-08-18T20:32:51Z |
| Radiomics/pyradiomics | 1459 | 549 | 2025-06-17T00:39:11Z |
| jalammar/ecco | 2101 | 175 | 2024-08-15T19:08:06Z |
| pyscf/pyscf | 1681 | 749 | 2026-09-28T05:45:43Z |

### World of Code (commits, authors, max time, min time)
| project | commits | authors | max time | min time |
|---|---|---|---|---|
| acellera_moleculekit | 1477 | 19 | 2025-10-20 11:34:40 | 2019-03-28 10:55:31 |
| druid-io_pydruid | 891 | 200 | 2025-10-09 19:32:30 | 2013-08-28 00:52:10 |
| florianhartig_dharma | 1281 | 45 | 2025-10-22 12:01:59 | 2015-02-10 15:43:53 |
| jalammar_ecco | 349 | 31 | 2025-09-29 19:20:49 | 2020-11-07 10:03:22 |
| molssi_qcengine | 2868 | 98 | 2025-11-05 16:40:37 | 2018-03-01 22:25:36 |
| nwfsc-cb_ocnms | 532 | 13 | 2025-08-14 20:32:09 | 2015-11-16 20:53:30 |
| projectchrono_chrono | 23025 | 266 | 2025-11-05 14:05:06 | 2010-02-23 11:03:14 |
| pyscf_pyscf | 14445 | 356 | 2025-11-05 21:06:40 | 2014-05-02 18:42:26 |
| radiomics_pyradiomics | 1679 | 58 | 2025-10-27 22:04:30 | 2015-09-15 15:19:31 |
| skuschel_postpic | 1229 | 21 | 2025-03-18 09:51:30 | 2013-12-05 19:43:54 |